rotkehlchen/chain/evm/decoding/paraswap/decoder.py

```diff
@@ -0,0 +1,203 @@
+                event.address = self.router_address
+                out_event = event
+            elif (
+                event.event_subtype == HistoryEventSubType.RECEIVE or
+                event.event_type == HistoryEventType.RECEIVE
+            ):  # find the receive event
+                event.counterparty = CPT_PARASWAP
+                event.event_type = HistoryEventType.TRADE
+                event.event_subtype = HistoryEventSubType.RECEIVE
+                event.notes = f'Receive {event.balance.amount} {event.asset.resolve_to_asset_with_symbol().symbol} as the result of a swap in paraswap'  # noqa: E501
+                event.address = self.router_address
+                if in_event is None:
+                    in_event = event
+                else:  # if another in_event is found, then it is the return of in_asset
+                    partial_refund_event = event
+
+        if in_event is None or out_event is None:
+            log.error(f'Could not find the corresponding events when decoding paraswap swap {context.transaction.tx_hash.hex()}')  # noqa: E501
+            return DEFAULT_DECODING_OUTPUT
+
+        if partial_refund_event is not None:  # if some in_asset is returned back
+            out_event.balance.amount -= partial_refund_event.balance.amount  # adjust the amount
+            context.decoded_events.remove(partial_refund_event)  # and remove it from the list
+        out_event.notes = f'Swap {out_event.balance.amount} {out_event.asset.resolve_to_asset_with_symbol().symbol} in paraswap'  # noqa: E501
+
+        maybe_reshuffle_events(
+            ordered_events=[out_event, in_event],
+            events_list=context.decoded_events,
+        )
+
+        fee_raw = fee_asset = None  # extract the fee info
+        for context_log in context.all_logs:
+            if (
+                context_log.topics[0] == ERC20_OR_ERC721_TRANSFER and
+                hex_or_bytes_to_address(context_log.topics[2]) == self.fee_receiver_address
+            ):
+                fee_raw = hex_or_bytes_to_int(context_log.data)
+                fee_asset = in_event.asset.resolve_to_evm_token()
+                break
+
+        if fee_raw is None or fee_asset is None:
+            # if fee is not found then we have already updated the events so exit
+            return DEFAULT_DECODING_OUTPUT
+
+        # update the in_event to adjust its balance since the amount used in fees
+        # was also received as part of the swap
+        fee_amount = asset_normalized_value(amount=fee_raw, asset=fee_asset)
+        if in_event.asset == fee_asset:
+            in_event.balance.amount += fee_amount
+            in_event.notes = f'Receive {in_event.balance.amount} {fee_asset.symbol} as the result of a swap in paraswap'  # noqa: E501
+
+        # And now create a new event for the fee
+        fee_event = self.base.make_event_from_transaction(
+            transaction=context.transaction,
+            tx_log=context.tx_log,
```

Comment:
we discussed about reshuffle with the fee event instead of propagating it to ensure the order